apps/src/templates/sectionProgress/sectionProgressRedux.js

```diff
@@ -109,6 +124,23 @@ export default function sectionProgress(state = initialState, action) {
       },
     };
   }
+  if (action.type === EXPAND_METADATA_FOR_STUDENTS) {
+    return {
+      ...state,
+      expandedMetadataStudentIds: [
+        ...state.expandedMetadataStudentIds,
```

Comment:
Do you intend a use case to add/remove multiple students at once? Perhaps an "expand all" button somewhere? It seems to be only one at a time in the current implementation though. 

Regardless it might be useful to ensure that duplicates can't be added here.

Reply:
1) Yep, we want an expand all and collapse all button.

2) I think duplicates should be fine because the collapse method removes all duplicates. Everything should just check whether an id is in the table. I think that the performance gain will not outweigh the cost of de-duplicating.

Reply:
The collapse action seems to only remove duplicates of whatever you're removing.

Consider a small section with student IDs `[1,2,3,4,5]`.

Slightly malicious scenario:
1. Expand all. `expandedMetadataStudentIds = [1, 2, 3, 4, 5]`
2. Collapse student 1. `expandedMetadataStudentIds = [2, 3, 4, 5]`
3. Expand all. `expandedMetadataStudentIds = [2, 3, 4, 5, 1, 2, 3, 4, 5]`
4. Collapse student 1. `expandedMetadataStudentIds = [2, 3, 4, 5, 2, 3, 4, 5]`
5. Expand all. `expandedMetadataStudentIds = [2, 3, 4, 5, 2, 3, 4, 5, 1, 2, 3, 4, 5]`
6. ...

Your point about premature optimization is well taken though. 👍

Reply:
Very good example. I added the deduplication